### PR TITLE
Proxy support

### PR DESCRIPTION
**Pull request by mad-mike**
implementing #25

### CODE REVIEW

File: src/models/configurationSettings.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-"use strict";
+/// <reference path="../../typings/vscode-typings.d.ts" />
```
Comment (Huachao) [inline src/models/configurationSettings.ts]: I have a question that why adding this line and removing the strict mode?

Comment (mad-mike) [inline src/models/configurationSettings.ts]: I found that the typescript compiler will introduce "use strict" anyway.
Then I added the ref to the type definitions in order to get IntelliSense for the vscode API.

Comment (Huachao) [inline src/models/configurationSettings.ts]: I found even without this line, I can still get IntelliSense with vscode

Comment (mad-mike) [inline src/models/configurationSettings.ts]: @Huachao, you are right. Explicit typings are only necessary for npm packages that don't contain them out of the box, as it seems (http://www.typescriptlang.org/docs/handbook/typings-for-npm-packages.html). I'm going to save a few lines of code from now on ;)

**Comment by Huachao** [comment]
Merged, thanks

**Pull request merged**